fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreBase.java

```diff
@@ -81,13 +88,19 @@ public SubspaceProvider getSubspaceProvider() {
     }
 
     @Nonnull
-    public Subspace getSubspace() {
-        return subspaceProvider.getSubspace(context);
+    public CompletableFuture<Subspace> getSubspaceAsync() {
+        if (subspaceFuture == null) {
+            subspaceFuture = subspaceProvider.getSubspaceAsync(context);
+        }
+        return subspaceFuture;
     }
 
     @Nonnull
-    public CompletableFuture<Subspace> getSubspaceAsync() {
-        return subspaceProvider.getSubspaceAsync(context);
+    public Subspace getSubspace() {
+        if (subspace == null) {
+            subspace = subspaceProvider.getSubspace(context);
+        }
```

Comment:
If you have a caller calling both the asynchronous and the synchronous versions of these methods, you will resolve the subspace twice.  It would probably make sense to use the cached future all the time, like:

    return context.asyncToSync(FDBStoreTimer.Waits.WAIT_KEYSPACE_PATH_RESOLVE,  getSubspaceAsync(context));

this way you will always only resolve once ever.